__tests__/ProfileImageList-test.js

```diff
@@ -0,0 +1,161 @@
+    const iconButtonWrapper = wrapper.find(IconButton);
+    const appsIconButtonWrapper = iconButtonWrapper.find({
+      icon: 'apps',
+    }); /** Next find the one that has the property icon grid. */
+    const appsIconButtonProps = appsIconButtonWrapper.first().props();
+    appsIconButtonProps.onPress(); /** Click the onPress function which changes the state. */
+    /** We only care that the sate is changing as needed.  */
+    const profileViewStateAfterClick =
+      profileImageListInstance.state.profile_view;
+    expect(profileViewStateAfterClick).toEqual('grid');
+  });
+
+  it('onViewHeadlineIconButtonClicked profile_view state is list', () => {
+    const wrapper = shallow(<ProfileImageList />);
+    /** Do we actually need the state before? It shouldn't be of concern to us. */
+    const profileImageListInstance = wrapper.instance(); /** To ensure they all use the same instance declare . */
+    const profileViewStateBeforeClick =
+      profileImageListInstance.state.profile_view;
+    const iconButtonWrapper = wrapper.find(IconButton);
+    const viewHeadlineIconButtonWrapper = iconButtonWrapper.find({
+      icon: 'view-headline',
+    }); /** Next find the one that has the property icon grid. */
+    const viewHeadlineIconButtonProps = viewHeadlineIconButtonWrapper
+      .first()
+      .props();
+    viewHeadlineIconButtonProps.onPress(); /** Click the onPress function which changes the state. */
+    /** We only care that the sate is changing as needed.  */
+    const profileViewStateAfterClick =
+      profileImageListInstance.state.profile_view;
+    expect(profileViewStateAfterClick).toEqual('list');
+  });
+}
+
+/** Test for which FlatList is rendered, depending on the profile view state */
+function testWhichFlatlistShows() {
+  it('Profile View state = grid and 3 column flatlist is present', () => {
+    const wrapper = shallow(<ProfileImageList />);
+    const instance = wrapper.instance();
+    instance.setState({
+      profile_view: 'grid',
+    }); /** WE just want to check the grid view first. (Force value) */
+    /** get the current state -- for what reason? */
+    //const viewState = instance.state.profile_view;
+    //expect(viewState).toEqual('grid');
+
+    const flatlistWrapper = wrapper.find(FlatList);
+    //const threeColWrapper = flatlistWrapper.find({numColumns: 3});
+    //console.log(threeColWrapper.first().props().numColumns);
+
+    expect(flatlistWrapper.find({numColumns: 3})).toHaveLength(1);
+  });
+
+  it('Profile View state = grid and 2 column flatlist is NOT present', () => {
+    const wrapper = shallow(<ProfileImageList />);
+    const instance = wrapper.instance();
+    instance.setState({
+      profile_view: 'grid',
+    }); /** WE just want to check the grid view first. (Force value) */
+    /** get the current state -- for what reason? */
+    //const viewState = instance.state.profile_view;
+    //expect(viewState).toEqual('grid');
```

Comment:
No need to add comments explainng each test. The name of the test should do that well enough